skbio/metadata/_repr.py

```diff
@@ -126,6 +127,15 @@ def _format_positional_metadata_column(self, key, dtype):
         dtype_fmt = '<dtype: %s>' % str(dtype)
         return self._wrap_text_with_indent(dtype_fmt, key_fmt, 1)
 
+    def _process_interval_metadata(self):
+        if self._obj.has_interval_metadata():
```

Comment:
To work around the Sequence/TabularMSA repr issues, add this:

```python
if hasattr(self._obj, "has_interval_metadata") and self._obj.has_interval_metadata():
    ...
```

This is a little gross, but this repr class (`_MetadataReprBuilder`) already needs some refactoring so that classes can mix and match which types of metadata they have and need reprs for (`.metadata`, `.positional_metadata`, `.interval_metadata`). It probably makes sense to remove this class and move each of the metadata repr helpers into their respective metadata mixin classes (e.g. as `_repr_` methods). This is a larger change that's not super important to make (it's all private API) so I think using the `hasattr` workaround is a practical way to move forward here.

It's also fine if you want to remove the repr updates (and mixin doctests) from this PR, and add them in the upcoming Sequence PR. I think we'll still need the `hasattr` workaround.